nltk/parse/bllip.py

```diff
@@ -100,11 +100,11 @@ def _ensure_bllip_import_or_error(ie=ie):
 def _ensure_ascii(words):
     try:
         for i, word in enumerate(words):
-            word.decode("ascii")
-    except UnicodeDecodeError as e:
+            word.encode("ascii")
+    except UnicodeEncodeError as e:
         raise ValueError(
-            "Token %d (%r) is non-ASCII. BLLIP Parser "
-            "currently doesn't support non-ASCII inputs." % (i, word)
+            f"Token {i} ({word}) is non-ASCII. BLLIP Parser "
```

Comment:
```suggestion
            f"Token {i} ({word!r}) is non-ASCII. BLLIP Parser "
```
I believe for it to be equivalent, you would want to get the `repr` of `word`. I haven't tested this for certain, but if I remember correctly, you can just do this with `{word!r}`. (For the record, I believe it's also fine to just print `word`)

Reply:
good catch! i agree -- while it probably doesn't matter, using the repr (`!r`) is slightly more in the spirit of this error message since it's helpful for debugging